src/sandstorm/run-bundle.c++

```diff
@@ -1270,6 +1313,23 @@ private:
       }
     }
 
+    // Now process the PORT setting, since the actual value in config.ports
+    // depends on if HTTPS_PORT was provided at any point in reading the
+    // config file.
+    //
+    // Outer KJ_IF_MAYBE so we only run this code if the config file contained
+    // a PORT= declaration.
+    KJ_IF_MAYBE(portValue, maybePortValue) {
+        KJ_IF_MAYBE(ports, getPorts(config.httpsPort, *portValue)) {
+            // Memory leak? Who owns this pointer? I guess it has to stay
```

Comment:
No memory leak here:

- getPorts() returned a `Maybe<Array<uint>>` which, as it turns out, contains (and owns) an Array<uint>.
- The `Maybe<Array<uint>>` is now on the stack. When it goes out of scope, it will be cleaned up by the destructor.
- `ports` points to the inner `Array<uint>` (only in the branch of `KJ_IF_MAYBE` where one exists).
- Below, you use `config.ports = kj::mv(*ports);` to transfer ownership of this array into `config.ports`. Now it will live as long as `config` does. It is no longer owned by the `Maybe` -- the `Maybe` now contains an unspecified `Array<uint>` value whose destructor will do nothing.